### PR TITLE
[SDK] Suppress config change detection based on placement rules

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

**Comment by mbdcos** [comment]
@gabrielhartmann Could you please explain what you are doing here? May be a few lines in description.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java [hunk 1]
```diff
@@ -175,11 +177,24 @@ private Comparator() {
         }
 
         @Override
-        public boolean equals(ServiceSpec first, ServiceSpec second) {
-            return EqualsBuilder.reflectionEquals(first, second);
+        public boolean equals(ServiceSpec current, ServiceSpec proposed) {
```
Comment (mbdcos) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java]: See: DefaultConfigurationUpdater.java:L117 
You can not change placement rule at all if this PR is in. Am I missing something here?

Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java]: Yep, you are right. Good catch.

**Comment by benclarkwood** [comment]
This PR would have the bad consequence of not letting you change placement constraints ever.

https://github.com/mesosphere/dcos-commons/pull/811 has the proper solution.

**Pull request closed without merging**